plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveDecimalToVarcharCoercion.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive;
+
+import com.google.common.collect.ImmutableList;
+import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.QueryRunner;
+import org.testng.annotations.Test;
+
+import static io.trino.testing.sql.TestTable.randomTableSuffix;
+import static io.trino.tpch.TpchTable.NATION;
+
+public class TestHiveDecimalToVarcharCoercion
+        extends AbstractTestQueryFramework
+{
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        return HiveQueryRunner.builder()
+                .setInitialTables(ImmutableList.of(NATION))
+                .build();
+    }
+
+    @Test
+    public void testShortDecimalToVarcharCoercion()
+    {
+        String tableName = "test_short_decimal_to_varchar_coercion_" + randomTableSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " (col1 DECIMAL(10,3), col2 int, col3 int) with ( FORMAT = 'PARQUET', partitioned_by = ARRAY['col3'])");
+        assertUpdate("INSERT INTO " + tableName + "(col1, col2, col3) VALUES(DECIMAL '100.13', 1, 1)", 1);
+        query("ALTER TABLE " + tableName + " DROP COLUMN col1");
+        query("ALTER TABLE " + tableName + " ADD COLUMN col1 VARCHAR");
+        assertUpdate("INSERT INTO " + tableName + " (col1, col2, col3) VALUES('1000.222', 2, 2)", 1);
+        assertQuery("SELECT col1, col2, col3 FROM " + tableName, "VALUES ('100.130', 1, 1), ('1000.222', 2, 2)");
+    }
+
+    @Test
+    public void testLongDecimalToVarcharCoercion()
+    {
+        String tableName = "test_long_decimal_to_varchar_coercion_" + randomTableSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " (col1 DECIMAL(23,3), col2 int, col3 int) with ( FORMAT = 'PARQUET', partitioned_by = ARRAY['col3'])");
+        assertUpdate("INSERT INTO " + tableName + "(col1, col2, col3) VALUES(DECIMAL '100.13', 1, 1)", 1);
+        query("ALTER TABLE " + tableName + " DROP COLUMN col1");
+        query("ALTER TABLE " + tableName + " ADD COLUMN col1 VARCHAR");
+        assertUpdate("INSERT INTO " + tableName + " (col1, col2, col3) VALUES('1000.222', 2, 2)", 1);
+        assertQuery("SELECT col1, col2, col3 FROM " + tableName, "VALUES ('100.130', 1, 1), ('1000.222', 2, 2)");
```

Comment:
Do we want `100.130` or would we rather have `100.13` in such a case?

Reply:
if you run select on like 53, just before tinkering with table format you will get `100.130` at leat with parquet file format. I am not planning to change how decimal are represented under the hood